### PR TITLE
Invoice list detail APIs

### LINKED ISSUE #37: Invoice list&detail APIs

### LINKED ISSUE #86: Add swagger to invoice's api

### PR DESCRIPTION
**Pull request by ziadnabil**
Invoice list create api
Screenshot from 2020-10-23 19-54-52

Invoice detail patch api

Screenshot from 2020-10-23 20-08-29

### CODE REVIEW

**Review by shahwan42 — DISMISSED** [review]

**Review by shahwan42 — CHANGES_REQUESTED** [review]
Add tests please

File: xshop/invoices/api/views.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from rest_framework.views import APIView
+from rest_framework import status
+from ..models import Invoice
+from .serializers import InvoiceSerializer
+
+
+class InvoiceListCreateApi(APIView):
+    serializer_class = InvoiceSerializer
+
+    def get(self, request):
```
Comment (mmostafa74) [inline xshop/invoices/api/views.py:15]: you need to add  @swagger_auto_schema to your endpoints to make it easier with listing schema and responses sent after each operation to list all those data in swagger-ui 
you can take a look here: https://drf-yasg.readthedocs.io/en/stable/custom_spec.html
and check 74_swagger_update branch

Comment (shahwan42) [reply xshop/invoices/api/views.py:15]: This PR shows how #74

File: xshop/invoices/tests/test_api/test_views.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def test_invoice_api_string_order(self):
+        invoice_data = {"user": 1, "order": "1"}
+        resp = self.client.post(self.url, invoice_data)
+        self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_retrieve_existing_invoice(self):
+        resp = self.client.get(self.detail_patch_url(self.invoice1.id))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(resp.data["user"], self.user.id)
```
Comment (shahwan42) [inline xshop/invoices/tests/test_api/test_views.py]: assert same invoice's id please

File: xshop/pages/api/views.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class LoginTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+        self.url = reverse("login")
+
+    def test_login_uses_desired_template(self):
+        resp = self.client.get(self.url)
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertTemplateUsed(resp, "registration/login.html")
```
Comment (shahwan42) [inline xshop/pages/api/views.py]: Why this here? these are tests, when in api/views file?

File: xshop/pages/tests/test_views.py [hunk 4]
```diff
@@ -1,28 +0,0 @@
...
-class LoginTests(TestCase):
-    def setUp(self) -> None:
-        self.client = Client()
-        self.url = reverse("login")
-
-    def test_login_uses_desired_template(self):
-        resp = self.client.get(self.url)
-
-        self.assertEqual(resp.status_code, status.HTTP_200_OK)
-        self.assertTemplateUsed(resp, "registration/login.html")
```
Comment (shahwan42) [inline xshop/pages/tests/test_views.py]: Why these were removed from here?

**Review by shahwan42 — CHANGES_REQUESTED** [review]

**Review by shahwan42 — CHANGES_REQUESTED** [review]
add swagger b2a xD

**Review by shahwan42 — APPROVED** [review]

**Pull request merged**